### PR TITLE
Open source LLM guide

### PR DESCRIPTION
**Pull request by rlancemartin**
Guide for using open source LLMs locally.

### CODE REVIEW

File: docs/extras/use_cases/local_llms.ipynb [hunk 1]
```diff
@@ -0,0 +1,848 @@
...
+    "\n",
+    "In general, these frameworks will do a few things:\n",
+    "\n",
+    "1. `Quantization`: Reduce the memory footprint of the model weights\n",
+    "2. `Efficient implementation for inference`: Run inference quickly on consumer hardware\n",
+    "\n",
+    "\n",
+    "## Quickstart\n",
+    "\n",
+    "[`Ollama`](https://ollama.ai/) is one way to easily run inference on Apple silicon.\n",
```
Comment (jmorganca) [inline docs/extras/use_cases/local_llms.ipynb:59]:
```suggestion
    "[`Ollama`](https://ollama.ai/) is one way to easily run inference on macOS.\n",
```

File: docs/extras/use_cases/local_llms.ipynb [hunk 2]
```diff
@@ -0,0 +1,848 @@
...
+    "\n",
+    "There are various ways to gain access to model weights.\n",
+    "\n",
+    "1. [`HuggingFace`](https://huggingface.co/TheBloke) - Many quantized model are available for download and can be run with framework such as [`llama.cpp`](https://github.com/ggerganov/llama.cpp)\n",
+    "2. [`gpt4all`](https://gpt4all.io/index.html) - The model explorer offers a leaderboard of metrics and associated quantized models available for download \n",
+    "3. [`Ollama`](https://github.com/jmorganca/ollama) - Several models can be accessed directly \n",
+    "\n",
+    "### Ollama\n",
+    "\n",
+    "With [Ollama](docs/integrations/llms/ollama), fetch a model via `ollama pull <model family>:<model version>`:\n",
```
Comment (jmorganca) [inline docs/extras/use_cases/local_llms.ipynb:187]:
```suggestion
    "With [Ollama](docs/integrations/llms/ollama), fetch a model via `ollama pull <model>:<tag>`:\n",
```

File: docs/extras/use_cases/local_llms.ipynb [hunk 3]
```diff
@@ -0,0 +1,848 @@
...
+    "2. [GPT4All](https://gpt4all.io/index.html)\n",
+    "3. [HuggingFace](https://huggingface.co/spaces/lmsys/chatbot-arena-leaderboard)\n",
+    "\n",
+    "### Inference\n",
+    "\n",
+    "A few frameworks for this have emerged to support inference of open source LLMs on various devices (e.g., laptops):\n",
+    "\n",
+    "1. [`llama.cpp`](https://github.com/ggerganov/llama.cpp): C++ implementation of llama inference code with [weight optimization / quantization](https://finbarr.ca/how-is-llama-cpp-possible/)\n",
+    "2. [`gpt4all`](https://docs.gpt4all.io/index.html): Optimized C backend for inference\n",
+    "3. [`Ollama`](https://ollama.ai/): Bundling of model weights and enviornment into an app that runs on device and serves the LLM \n",
```
Comment (jmorganca) [inline docs/extras/use_cases/local_llms.ipynb:49]:
```suggestion
    "3. [`Ollama`](https://ollama.ai/): Bundling of model weights and environment into an app that runs on device and serves the LLM \n",
```

File: docs/extras/use_cases/local_llms.ipynb [hunk 4]
```diff
@@ -0,0 +1,848 @@
...
+    "\n",
+    "To minimize latency, users often want to run models locally on GPU.\n",
+    "\n",
+    "Some consumer laptops ship with GPUs, [such Apple devices](https://www.apple.com/newsroom/2022/06/apple-unveils-m2-with-breakthrough-performance-and-capabilities/).\n",
+    "\n",
+    "### Running Apple silicon GPU\n",
+    "\n",
+    "Ollama will automatically utilize the GPU on Apple devices.\n",
+    " \n",
+    "Other frameworks require the user to set up the enviornment to utilize the Apple GPU.\n",
```
Comment (jmorganca) [inline docs/extras/use_cases/local_llms.ipynb:149]:
```suggestion
    "Other frameworks require the user to set up the environment to utilize the Apple GPU.\n",
```

File: docs/extras/use_cases/local_llms.ipynb [hunk 5]
```diff
@@ -0,0 +1,848 @@
...
+      ]
+     },
+     "execution_count": 34,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from langchain.llms import Ollama\n",
+    "llm = Ollama(base_url=\"http://localhost:11434\",model=\"llama2\")\n",
```
Comment (jmorganca) [inline docs/extras/use_cases/local_llms.ipynb:87]:
```suggestion
    "llm = Ollama(model=\"llama2\")\n",
```

File: docs/extras/use_cases/local_llms.ipynb [hunk 6]
```diff
@@ -0,0 +1,848 @@
...
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from langchain.callbacks.manager import CallbackManager\n",
+    "from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler \n",
+    "llm = Ollama(base_url=\"http://localhost:11434\", \n",
+    "             model=\"llama2\", \n",
+    "             callback_manager = CallbackManager([StreamingStdOutCallbackHandler()]))\n",
```
Comment (jmorganca) [inline docs/extras/use_cases/local_llms.ipynb:128]:
```suggestion
    "llm = Ollama(model=\"llama2\", \n",
    "             callback_manager = CallbackManager([StreamingStdOutCallbackHandler()]))\n",
```

**Comment by baskaryan** [comment]
was imagining this would go in Guides rather than Use cases, wdyt @rlancemartin?

**Comment by rlancemartin** [comment]
> was imagining this would go in Guides rather than Use cases, wdyt @rlancemartin?

Ah yes, forgot abt `guides` section. Moving now!

**Pull request merged**